controller/replica_controller.go

```diff
@@ -303,11 +303,18 @@ func (rc *ReplicaController) syncReplica(key string) (err error) {
 			return errors.Wrapf(err, "failed to cleanup the related replica instance before deleting replica %v", replica.Name)
 		}
 
+		rs, err := rc.ds.ListReplicasByNodeRO(replica.Spec.NodeID)
```

Comment:
nit: rs can be initialized later in the following `else if replica.Spec.NodeID != ""` condition.